optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -144,7 +144,7 @@ def _metric_with_eval_at(self, metric: str) -> str:
     def higher_is_better(self) -> bool:
 
         metric_name = self.lgbm_params.get("metric", "binary_logloss")
-        return metric_name.startswith(("auc", "ndcg", "map"))
+        return metric_name in ("auc", "ndcg", "map")
```

Comment:
Orthogonal to this change, but a related comment. Should be add an assertion before this line to make sure we've in control of all possible values of `metric_name`, i.e. `assert metric_name in (...)`?

Reply:
Looks good. I agree with @hvy.

Reply:
Hmm, I think it is reasonable, but I feel it is out-of-scope of this PR.

LightGBM is being developed actively and we can see some new metrics like [`gamma` and `tweedie`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric-parameters) since September (https://github.com/optuna/optuna/issues/1806#issuecomment-694243823). Such new metrics (`gamma` and `tweedie`) can be handled properly now since they are to be minimized, but the assertion errors will be raised if we apply your suggestion. It may be controversial and I'd like to discuss it in a new PR.

Reply:
Sounds good to me.

Reply:
Thanks for the explanation.

> It may be controversial and I'd like to discuss it in a new PR.

It's reasonable. Thanks.